crates/storage/src/codec.rs

```diff
@@ -16,9 +16,12 @@ pub mod primitive;
 pub mod raw;
 
 /// The trait is usually implemented by the encoder that stores serialized objects.
-pub trait Encoder {
+pub trait Encoder<'a> {
     /// Returns the serialized object as a slice.
     fn as_bytes(&self) -> Cow<[u8]>;
+
+    /// Consumes the encoder, returning the serialized object as a slice.
+    fn into_bytes(self) -> Cow<'a, [u8]>;
```

Comment:
https://github.com/FuelLabs/fuel-core/pull/2420

Reply:
Nice, approved and merged :)